Removed redundant `data` parameter on Request's `transformData(_:serviceSuccess:)` method.

tylermilner opened the pull request:
This is some experimentation with what was discussed in #69. The unit tests and documentation still needs to be updated if we decide to implement this.

wmcginty commented:
This looks good to me. I like how it's basically the best of both worlds. All the 'default' implementations can still work normally and operate on `Data` (allowing them to be composed into the function if you don't want to modify that direct functionality) and you also get the ability to operate on the `HTTP.Response` if you need to.

Two thumbs up here.

wmcginty commented on Sources/Hyperspace/Request/Decoding/DecodableContainer.swift:
```diff
@@ -31,9 +31,9 @@ extension AnyRequest where T: Decodable {
-                return try .success(decoder.decode(U.ContainedType.self, from: data, with: U.self))
+                return try .success(decoder.decode(U.ContainedType.self, from: success.data, with: U.self))
             } catch {
                 return .failure(AnyError(error))
```
Total side note, but I think now with the introduction of `DecodingFailureInitializable` this can be condensed into `Request`. I'll take a look at doing that, I have some other work I'd like to do with error handling.

tylermilner commented:
@wmcginty with regards to the `2.0.0` adopters, how strict do you think we should be with semantic versioning? I'm torn between if this should be another major version update or just an update to the minor version number.

wmcginty commented:
@tylermilner I don't think we need to be overly strict - for instance, technically 2.1 could (and did in some cases) cause compiler errors in update. Now that 2.1 is out the door, we should probably stick to the rules of thumb we've been using.

tylermilner commented:
@wmcginty @earlgaspard I fixed the unit tests and updated the documentation so I think this should be good to merge into `master` now. Take a look whenever you get a chance.
